src/main/java/sg/ncl/MainController.java

```diff
@@ -2177,13 +2179,19 @@ public String startExperiment(
 
         // ensure experiment is stopped first before starting
         Realization realization = invokeAndExtractRealization(teamName, Long.parseLong(expId));
+        String teamStatus = getTeamStatus(realization.getTeamId());
 
         if (!checkPermissionRealizeExperiment(realization, session)) {
             log.warn("Permission denied to start experiment: {} for team: {}", realization.getExperimentName(), teamName);
             redirectAttributes.addFlashAttribute(MESSAGE, permissionDeniedMessage);
             return "redirect:/experiments";
         }
 
+        if (!teamStatus.equals(TeamStatus.APPROVED.name())) {
+            log.warn("Error: trying to realize an experiment {} on team {} with status {}", realization.getExperimentName(), realization.getId(), teamStatus);
+            redirectAttributes.addFlashAttribute(MESSAGE, "Team " + teamName + " is currently " + teamStatus + " and does not have permission to start experiment " + realization.getExperimentName() + " Please contact " + CONTACT_EMAIL);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Team " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)